plugin/src/index.ts

```diff
@@ -1,12 +1,24 @@
 import { registerPlugin } from '@capacitor/core';
 
-import type { PortalsPlugin } from './definitions';
+import type { InitialContext, PortalsPlugin } from './definitions';
 
 const Portals = registerPlugin<PortalsPlugin>('Portals', {
   web: () => import('./web').then(m => new m.PortalsWeb()),
   android: () => import('./android').then(m => new m.PortalsAndroid()),
   ios: () => import('./ios').then(m => new m.PortalsIOS())
 });
 
+export function getInitialContext<T = unknown>(): InitialContext<T> {
+  const context = (window as any).portalInitialContext as { name: string, value: T; };
+  if (!context) {
+    throw Error('No initial context found');
```

Comment:
@IT-MikeS or @eric-horodyski Do y'all think this should throw or just return undefined?

Reply:
My opinion: if `<T>` is defined and `!(context instanceof T)` throw an error. If `<T>` isn't defined, return `undefined`.